internal/cli/atlas/clusters/upgrade.go

```diff
@@ -0,0 +1,153 @@
+
+package clusters
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/file"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const upgradeTemplate = "Upgrading cluster '{{.Name}}'.\n"
+
+type UpgradeOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	name       string
+	mdbVersion string
+	diskSizeGB float64
+	tier       string
+	filename   string
+	fs         afero.Fs
+	store      store.AtlasSharedClusterGetterUpgrader
+}
+
+func (opts *UpgradeOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *UpgradeOpts) Run() error {
+	cluster, err := opts.cluster()
+	if err != nil {
+		return err
+	}
+	if opts.filename == "" {
+		opts.patchOpts(cluster)
+	}
+
+	r, err := opts.store.UpgradeCluster(opts.ConfigProjectID(), cluster)
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+func (opts *UpgradeOpts) cluster() (*atlas.Cluster, error) {
+	var cluster *atlas.Cluster
+	if opts.filename != "" {
+		err := file.Load(opts.fs, opts.filename, &cluster)
+		if err != nil {
+			return nil, err
+		}
+		if opts.name == "" {
+			opts.name = cluster.Name
+		}
+		return cluster, nil
+	}
+	return opts.store.AtlasSharedCluster(opts.ProjectID, opts.name)
+}
+
+func (opts *UpgradeOpts) patchOpts(out *atlas.Cluster) {
+	RemoveReadOnlyAttributesSharedCluster(out)
+	if opts.mdbVersion != "" {
+		out.MongoDBMajorVersion = opts.mdbVersion
+	}
+	if opts.diskSizeGB > 0 {
+		out.DiskSizeGB = &opts.diskSizeGB
+	}
+	if opts.tier != "" {
+		if out.ProviderSettings != nil {
+			out.ProviderSettings.InstanceSizeName = opts.tier
+			if opts.tier != "M2" && opts.tier != "M5" {
+				out.ProviderSettings.ProviderName = out.ProviderSettings.BackingProviderName
+				out.ProviderSettings.BackingProviderName = ""
+			}
+		}
+	}
+	AddLabelSharedCluster(out, atlas.Label{
+		Key:   labelKey,
+		Value: labelValue,
+	})
+}
+
+// mongocli atlas cluster(s) upgrade [clusterName] --projectId projectId [--tier M#] [--diskSizeGB N] [--mdbVersion].
+func UpgradeBuilder() *cobra.Command {
+	opts := UpgradeOpts{
+		fs: afero.NewOsFs(),
+	}
+	cmd := &cobra.Command{
+		Use:   "upgrade [clusterName]",
+		Short: "Upgrade a MongoDB Atlas cluster.",
+		Example: fmt.Sprintf(`  Upgrade tier, disk size and MongoDB version for a cluster:
+  $ %s cluster upgrade <clusterName> --projectId <projectId> --tier M50 --diskSizeGB 20 --mdbVersion 4.2`,
+			cli.ExampleAtlasEntryPoint()),
+		Args: require.MaximumNArgs(1),
```

Comment:
Is clusterName mandatory?

Reply:
Yes, it's required according to the API docs.

Reply:
so, make it required 
```suggestion
		Args: require.ExactArgs(1),
```